packages/shared/lib/core/layer-2/utils/parseLayer2Metadata.ts

```diff
@@ -0,0 +1,51 @@
+import { ILayer2TransferAllowanceMetadata } from '../interfaces'
+
+type Layer2Metadata = Omit<ILayer2TransferAllowanceMetadata, 'baseTokenAmount' | 'nativeTokens'>
+
+export function parseLayer2Metadata(metadata: string): Layer2Metadata {
+    try {
+        const parsedData = metadata ? JSON.parse(metadata) : {}
+        validate(parsedData)
+        const parsedMetadata: Layer2Metadata = {
+            senderContract: parsedData.senderContract,
+            targetContract: parsedData.targetContract,
+            contractFunction: parsedData.contractFunction,
+            gasBudget: parsedData.gasBudget,
+            ethereumAddress: parsedData.ethereumAddress,
+            forceOpenAccount: parsedData.forceOpenAccount,
+        }
+        return parsedMetadata
+    } catch (error) {
+        return undefined
+    }
+}
+
+function validate(data: Layer2Metadata): void {
+    if (data.senderContract && typeof data.senderContract !== 'string') {
+        throw 'Invalid senderContract'
```

Comment:
Let's follow the official syntax and only throw Errors. 
```suggestion
        throw new Error('Invalid senderContract')
```